internal/pkg/runtime/platform/vmware/vmware.go

```diff
@@ -83,7 +83,8 @@ func (v *VMware) ExternalIPs() (addrs []net.IP, err error) {
 // KernelArgs implements the runtime.Platform interface.
 func (v *VMware) KernelArgs() kernel.Parameters {
 	return []*kernel.Parameter{
-		kernel.NewParameter("console").Append("ttyS0"),
+		kernel.NewParameter("console").Append("tty0"),
```

Comment:
Let's scope this PR to just this. I think this needs a bit more thought on how to best get the config arg.

Reply:
We can't scope it to just this. Without config=guestinfo we can't boot.

Reply:
if we keep the old way in place, then when we parse  / query kernel args, we pull the first arg which is config=none. 
there's definitely room for the vmware platform userdata bit to be enhanced, but i think that's a bit out of scope. eh maybe not. we can drop `guestinfo` as an option for talos.config and just go based on platform. part of the buy-in when you select a platform ( ex, if i pick aws i'm going to use metadata service for aws )

Reply:
We can scope it to fixing the console output. For cloud providers I think it makes sense that you buy in to the platform’s metadata service, but VMware is different. It should be more flexible. Plus, setting this here flips everything we have done with the installer. The installer allows for the configuration of the config option. It should not be baked into the platform. Our official images are configured with same defaults, but the idea with the installer has always been to allow for configuring this option.

Reply:
i disagree. 
vmware isn't any different. if you want to run on vmware but don't want to use the baked in cloud provider bits, then set the platform to bare metal and customize until you are content. otherwise it should behave like all other platforms and use the default discovery method for userdata. 
ill push up another commit to revert the changes to cmd/install and loosen up the provider/vmware bits / deprecate the talos.config=guestinfo part.

Reply:
Allowing for more than one way to configure VMware was an intentional design decision made when the VMware provider was first implemented. I would like to keep it that way. The reason being is that even in places like AWS, you can boot bare metal machines, and you may want to supply the config another way, but still use the other platform features like discovering the external IPs and all the other magic we do. That is a bit of a contrived example, but when I first implemented the VMware platform, I was using an http server to serve the configs, and found it very useful to _not_ have to use the `guestinfo` option. The installer has the option to explicitly provide the config source, and I want to keep that. The idea has always been to provide sane images out of the box, but to not _force_ users into a particular workflow. If they want to generate images, it is a simple docker command, and it has all the flags to configure the image to your liking. Let's hold this PR until we can come up with a better solution.